src/resolver.ts

```diff
@@ -1199,14 +1223,42 @@ export class Resolver extends DiagnosticEmitter {
     /** How to proceed with eventualy diagnostics. */
     reportMode: ReportMode = ReportMode.REPORT
   ): Element | null {
-    // TODO
-    if (reportMode == ReportMode.REPORT) {
-      this.error(
-        DiagnosticCode.Operation_not_supported,
-        expression.range
-      );
+    let left = expression.left;
+    let right = expression.right;
+    let leftElement = this.resolveExpression(left, flow, contextualType, reportMode);
+    if (!leftElement) return null;
+    let leftType = (<TypedElement>leftElement!).type;
+    let rightElement = this.resolveExpression(right, flow, leftType, reportMode);
+    if (!rightElement) return null;
+    let rightType = (<TypedElement>rightElement!).type;
+    let commonType: Type | null;
+    if (commonType = Type.commonDenominator(leftType, rightType, true)) {
```

Comment:
There are some occasions where it doesn't work this way, like for shifts which always prefer the LHS. I think there should be a switch over all possible operators here that mimics what `compileExpression` etc. do. In this case, `binaryOperatorKindFromToken` can possibly be incorporated into the switch.